logstash-core/lib/logstash/runner.rb

```diff
@@ -432,6 +436,44 @@ def execute
     @log_fd.close if @log_fd
   end # def self.main
 
+  def checkLogConfigurationContainsJavascriptUsage
```

Comment:
@yaauie I've added this check, in case the log4j format is the old one it make Logstash exit. What I dont' like is the output generated at log:
```
[2020-12-15T11:59:45,127][WARN ][logstash.runner          ]          Found old log configuration that leverage removed JavaScript engine.
         Replace 'ScriptedFilter' section in your /home/andrea/workspace/logstash_andsel/config/log4j2.properties with 'avoid_pipelined_filter.type = PipelineRoutingFilter'

[2020-12-15T11:59:45,129][WARN ][logstash.runner          ]          Found old log configuration that leverage removed JavaScript engine.
         Replace 'Routing' appender in your /home/andrea/workspace/logstash_andsel/config/log4j2.properties with
         'appender.routing.type = PipelineRouting'
         'appender.routing.pipeline.type = RollingFile'
         'appender.routing.pipeline.name = appender-${ctx:pipeline.id}'
         'appender.routing.pipeline.fileName = /home/andrea/workspace/logstash_andsel/logs/pipeline_${ctx:pipeline.id}.log'
         'appender.routing.pipeline.filePattern = /home/andrea/workspace/logstash_andsel/logs/pipeline_${ctx:pipeline.id}.%i.log.gz'
         'appender.routing.pipeline.layout.type = PatternLayout'
         'appender.routing.pipeline.layout.pattern = [%d{ISO8601}][%-5p][%-25c] %m%n'
         'appender.routing.pipeline.policy.type = SizeBasedTriggeringPolicy'
         'appender.routing.pipeline.policy.size = 100MB'
         'appender.routing.pipeline.strategy.type = DefaultRolloverStrategy'
         'appender.routing.pipeline.strategy.max = 30'

[2020-12-15T11:59:45,133][ERROR][org.logstash.Logstash    ] java.lang.IllegalStateException: Logstash stopped processing because of an error: (SystemExit) exit
```

Do you have any advise about it?

Reply:
on LS7, this should be a non-exiting deprecation log, and the message should be changed to indicate that the logging appender they have selected is deprecated and will be unavailable in Logstash 8.

~~~
Logging configuration uses deprecated log appender, and must be resolved before upgrading to Logstash 8.
~~~

Reply:
If I understand correctly, on LS7 we log the warn on the deprecation logger saying that in LS8 is going to be removed and continue the execution.
On LS8 should we log the warn and stop the process?

Reply:
@yaauie I've updated this let me know if it's good enought